smdebug/tensorflow/keras.py

```diff
@@ -55,6 +55,7 @@ def __init__(
         self.tensor_refs_to_save_this_step = set()
         self._fetches_added = set()
         self.callable_cache = CallableCache()
+        self.tape = None
```

Comment:
Since this variable is referred to in the base hook, it should be set in the base hook constructor. Otherwise the base hook method should be moved out to this class.